.github/workflows/cancel.yml

```diff
@@ -0,0 +1,14 @@
+name: Cancel
```

Comment:
I think this could be useful. but ignore it for now. I'll create another PR to include this if we like it.

Reply:
I'm in favour of adding this, our CI is heavy, any savings in terms of unnecessary CI time are welcomed both by us and probably GitHub too. I'm sure they're planning to have this built-in at some point, but in the mean time a separate action seems great!

Can you create the PR?

Reply:
Sure. It's coming.